qemu/tests/check_coredump.py

```diff
@@ -148,51 +148,69 @@ def run(test, params, env):
     4) If yes, copy them to working directory.
     """
 
+    error_msgbox = []
+    error.context("Check crash/coredump in test result dir", logging.info)
+    root_dir = os.path.dirname(test.resultsdir)
+    crash_dir = []
+    for prefix in ["crash", "core"]:
+        pattern = "%s/*/%s.*" % (root_dir, prefix)
+        crash_dir += glob.glob(pattern)
+    if crash_dir:
+        crash_msg = ("Crash/Coredump happened, " +
+                     "Please check dump files under %s" % crash_dir)
+        error_msgbox.append(crash_msg)
+
```

Comment:
On 05/12/2015 05:07 PM, suqinhuang wrote:
>
> In qemu/tests/check_coredump.py 
> <https://github.com/autotest/tp-qemu/pull/355#discussion_r30116380>:
>
> > @@ -148,51 +148,69 @@ def run(test, params, env):
> >      4) If yes, copy them to working directory.
> >      """
> >
> > +    error_msgbox = []
> > +    error.context("Check crash/coredump in test result dir", logging.info)
> > +    root_dir = os.path.dirname(test.resultsdir)
> > +    crash_dir = []
> > +    for prefix in ["crash", "core"]:
> > +        pattern = "%s/*/%s.*" % (root_dir, prefix)
> > +        crash_dir += glob.glob(pattern)
> > +    if crash_dir:
> > +        crash_msg = ("Crash/Coredump happened, " +
> > +                     "Please check dump files under %s" % crash_dir)
> > +        error_msgbox.append(crash_msg)
> > +
>
> test still pass when there is core in the result
>
> 17:03:21 INFO | GOOD 
> virt.qemu.smp_2.2048m.repeat1.post_check.Host_RHEL.7.1.qcow2.virtio_blk.up.virtio_net.RHEL.7.1.x86_64.io-github-autotest-qemu.check_coredump 
> virt.qemu.smp_2.2048m.repeat1.post_check.Host_RHEL.7.1.qcow2.virtio_blk.up.virtio_net.RHEL.7.1.x86_64.io-github-autotest-qemu.check_coredump 
> timestamp=1431421401 localtime=May 12 17:03:21 completed successfully
>
> -rw-r--r-- 1 root root 0 May 12 17:02 core
> drwxr-xr-x 4 root root 4096 May 12 17:03 debug
> -rw-r--r-- 1 root root 19123 May 12 17:03 keyval
> drwxr-xr-x 3 root root 24 May 12 17:03 profiling
> drwxr-xr-x 2 root root 6 May 12 17:02 results
> -rw-r--r-- 1 root root 964 May 12 17:03 status
> drwxr-xr-x 3 root root 87 May 12 17:03 sysinfo
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/autotest/tp-qemu/pull/355/files#r30116380>.
>
Hi shuang,

core pattern in crashhandler is core.xxx or crash.xxx,  Please test with 
core.xx.

Thanks,
Xu